src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -211,65 +211,41 @@ public void delete(final String eventTypeName, final Client client)
             eventType.setDefaultStatistic(
                     validateStatisticsUpdate(original.getDefaultStatistic(), eventType.getDefaultStatistic()));
-            final Long newRetentionTime = eventTypeBase.getOptions().getRetentionTime();
-            final Long oldRetentionTime = original.getOptions().getRetentionTime();
-            if (oldRetentionTime == null) {
-                // since we have some inconsistency in DB I will put here for a while
-                throw new InconsistentStateException("Empty value for retention time in existing EventType");
-            }
-            boolean retentionTimeUpdated = false;
-            try {
-                if (newRetentionTime != null && !newRetentionTime.equals(oldRetentionTime)) {
-                    updateTopicRetentionTime(eventTypeName, newRetentionTime);
-                } else {
-                    eventType.setOptions(original.getOptions());
-                }
-                updateEventTypeInDB(eventType, newRetentionTime, oldRetentionTime);
-                retentionTimeUpdated = true;
-            } finally {
-                if (!retentionTimeUpdated) {
-                    updateTopicRetentionTime(eventTypeName, oldRetentionTime);
-                }
-            }
-
-            return Result.ok();
+            updateRetentionTime(original, eventType);
         } catch (final InterruptedException e) {
             Thread.currentThread().interrupt();
-            LOG.error("Failed to wait for timeline switch", e);
-            return Result.problem(Problem.valueOf(Response.Status.SERVICE_UNAVAILABLE,
-                    "Event type is currently in maintenance, please repeat request"));
+            throw new ServiceTemporaryUnavailableException(
+                    "Event type is currently in maintenance, please repeat request", e);
         } catch (final TimeoutException e) {
             LOG.error("Failed to wait for timeline switch", e);
-            return Result.problem(Problem.valueOf(Response.Status.SERVICE_UNAVAILABLE,
-                    "Event type is currently in maintenance, please repeat request"));
-        } catch (final InvalidEventTypeException e) {
-            return Result.problem(e.asProblem());
-        } catch (final NoSuchEventTypeException e) {
-            LOG.debug("Could not find EventType: {}", eventTypeName);
-            return Result.problem(e.asProblem());
-        } catch (final NoSuchPartitionStrategyException e) {
-            LOG.debug("Partition strategy does not exist", e);
-            return Result.problem(e.asProblem());
+            throw new ServiceTemporaryUnavailableException(
+                    "Event type is currently in maintenance, please repeat request", e);
         } catch (final NakadiException e) {
             LOG.error("Unable to update event type", e);
-            return Result.problem(e.asProblem());
+            throw new NakadiRuntimeException(e);
```

Comment:
Which details? it will be unpacked in ExceptionHandling creating the same result